config/environments/production.rb

```diff
@@ -82,8 +82,9 @@
     :password => ENV['SENDGRID_PASSWORD'],
     :address => "smtp.sendgrid.net",
     :port => 587,
+    :domain  => 'yourapp.heroku.com',
```

Comment:
I am reading up on the Send Grid rules regarding hoe specific the URL can be if through heroku.